lib/src/node.c

```diff
@@ -513,7 +518,7 @@ TSNode ts_node_parent(TSNode self) {
         ts_node_start_byte(child) > ts_node_start_byte(self) ||
         child.id == self.id
       ) break;
-      if (iterator.position.bytes >= end_byte) {
+      if (iterator.position.bytes >= end_byte && !ts_node__subtree(child).data.is_inline) {
```

Comment:
I don't think the `is_inline` bit is the right criteria for this. The `is_inline` thing is just an optimization where certain leaf nodes are stored w/o heap allocations. Leaf nodes could be inlined or not based on the numerical values of their `size` and `padding` (e.g. the size needs to be small enough to fit in a certain number of bits).

Maybe we should be checking if `child` has children here (`ts_subtree_child_count(node) > 0`)? Or is that we need to skip empty nodes (where `ts_subtree_size(node).bytes > 0`)?

Reply:
Yeah `ts_subtree_child_count(node) > 0` is better actually, `ts_subtree_size(node).bytes > 0` doesn't work since the child is 0 bytes as well as the inlined parent